design-proposals/platform-installer-simplification.md

```diff
@@ -0,0 +1,441 @@
+AWS install, with the goal that as documentation of the AWS installation improves, the documentaiton will
+eventually become self-sufficient and a sufficiently skilled partner will be able to set up the AWS
+infrastructure with minimum assistance required.
+
+#### Migrate any remaining Helm/Kubernetes services from pre-installer to the installer
+
+The 3.1 installers may contain some helm-based components, such as Postgres, or other services that were
+installed prior to the invocation of ArgoCD. These should be moved from these pre-installers into the
+installer and handled by ArgoCD. We should have a consistent mechanism for installing Helm charts.
+
+Redundant components in can always be disabled. For example, if a cloud-based database such as Aurora is
+used, then we will have a knob that disables installation of Postgres.
+
+#### Ensure all pre-installers and the installer are noninteractive
+
+The pre-installers and the installer should be fully configured from the configuration values that are passed
+to them initially. There should be no interactive prompts given to the user and no pausing of the installer
+to solicit additional input.
+
+### Phase 2 - Composable Installer
+
+In the prior phase, we moved terraform recipes to the pre-installers only, but if for any reason terraform files
+still exist within the installer, we will move them in this phase. We continue to remove shell scripts and replace
+them with standard helm charts and Kubernetes jobs, all of which are exposed as ArgoCD apps. Pre-installers may
+continue to be used internally or distributed as examples.
+
+By the end of this phase, it shall be possible to invoke the installer directly, by the customer bringing their
+own Kubernetes, without using a pre-installer.
+
+The following need to be taken care to support this model
+
+- All Kubernetes objects including namespaces, databases, secrets, configs, policies etc must be treated as
+  application components. They must be created via Helm and Argo
+
+- Expose proper helm value overrides for customers to provide configuration values as appropriate
+
+#### Make subsystems easy to disable
+
+Make sure the installer configuration allows the following to be individually disabled, to support reduced footprint
+for customers who do not require all components. This will allow the following services and their pods to be disabled,
+reducing the footprint of the platform:
+
+- Observability
+
+  - Telemetry pods
+
+  - Grafana
+
+  - Loki
+
+  - Mimir
+
+- Cluster Orchestration
+
+  - cluster-connect-gateway
+
+  - cluster-manager
+
+  - intel-infra-provider
+
+  - capi-core-provider
+
+  - cluster-api-k3s-provider
+
+- Application Orchestration
+
+  - app-deployment
+
+  - app-interconnect
+
+  - app-orch-catalog
+
+  - app-orch-tenant-controller
+
+  - app-resource-manager
+
+  - app-service-proxy
+
+  - fleet
+
+  - gitea
+
+#### Add optional single-tenant initialization job
```

Comment:
There is one already - you can re-use: https://github.com/open-edge-platform/infra-charts/tree/main/tenant-config

Reply:
Thanks! I will update the text that we can re-use that job.